containerd: remove unleaseSnapshotsFromDeletedConfigs

jsternberg opened the pull request:
Removes workaround for https://github.com/moby/buildkit/issues/3797 now
that the underlying issue is fixed.

Fixes #46858.

vvoland reviewed (commented):
Thanks, looks good! Could we also have an integration test to make sure it doesn't regress in future?

jsternberg commented:
I added an integration test that builds an image and then deletes it immediately to check that it gets the deleted response. I was unable to figure out a way to check if the history item was still there but I think this should be good enough to check that it's working properly.

vvoland commented on integration/build/build_test.go:
```diff
@@ -793,6 +795,43 @@ func TestBuildEmitsImageCreateEvent(t *testing.T) {
+	err := buildImage("target-a")
+	assert.NilError(t, err)
+
+	resp, err := apiClient.ImageRemove(ctx, "target-a", image.RemoveOptions{})
+	assert.NilError(t, err)
+	assert.Check(t, slices.ContainsFunc(resp, func(r image.DeleteResponse) bool {
+		return r.Deleted != ""
+	}))
+}
```
IIRC the issue was that the image index was deleted just fine (so it would get reported in the `Deleted`) - it was only the actual blobs/snapshots that weren't GC'd due to buildkit holding the gc labels.

I think we could just make this test c8d-integration only, and then check with `ctr` if the content is indeed gone at this point.

(We probably need to also prune build cache before this?)

jsternberg replied:
I'm not sure if I tested it wrong, but my testing indicated that checking for deleted signified it was fixed: https://github.com/moby/moby/issues/46858#issuecomment-2822072127

The previous version only showed `Untagged`.

vvoland replied:
That's also possible - if you checked and that's indeed the case then testing the content store directly is a nice improvement, but not really a blocker.

Let me know if you want to try that.

jsternberg committed:
containerd: remove unleaseSnapshotsFromDeletedConfigs

Removes workaround for https://github.com/moby/buildkit/issues/3797 now
that the underlying issue is fixed.

Signed-off-by: Jonathan A. Sternberg <jonathan.sternberg@docker.com>

vvoland commented:
Needs rebase

jsternberg commented:
Rebased.
